test/langtools/tools/javac/modules/AnnotationsOnModules.java

```diff
@@ -726,6 +729,96 @@ public TestCase(String extraDecl, String decl, String use, String expectedAnnota
         }
     }
 
+    @Test
+    public void testBrokenModuleInfoClassWithAnnotation(Path base) throws Exception {
+        Path lib = base.resolve("lib");
+        tb.writeJavaFiles(lib,
+                          """
+                          @Deprecated
+                          module m{}
+                          """);
+
+        Path libClasses = base.resolve("lib-classes");
+        Files.createDirectories(libClasses);
+
+        new JavacTask(tb)
+            .options("--release", "21")
+            .outdir(libClasses)
+            .files(findJavaFiles(lib))
+            .run()
+            .writeAll();
+
+        Path modifiedModuleInfo = libClasses.resolve("module-info.class");
+        ClassModel cm1 = ClassFile.of().parse(modifiedModuleInfo);
+        byte[] newBytes = ClassFile.of().transformClass(cm1, new ClassTransform() {
+            @Override
+            public void accept(ClassBuilder builder, ClassElement element) {
```

Comment:
Is there any restriction so that you cannot use a lambda here?

Reply:
No, fixed.